scipy/integrate/_quadrature.py

```diff
@@ -662,6 +664,254 @@ def simpson(y, x=None, dx=1.0, axis=-1, even='avg'):
+    )
+
+    # Addition of above formulae gives Simpson's 1/3 rule for unequal intervals, see
+    # https://en.wikipedia.org/wiki/Simpson%27s_rule#Composite_Simpson's_rule_for_irregularly_spaced_data
+
+    res = _cumulatively_sum_simpson_subintegrals(
+        sub_integrals_h1,
+        sub_integrals_h2,
+    )
+    return res.swapaxes(0, axis)
+
+
+def cumulative_simpson(y, x=None, dx=1.0, axis=-1, initial=None):
+    """
+    Cumulatively integrate y(x) using the composite Simpson's 1/3 rule.
+    If x is None, spacing of dx is assumed.
+
+    The integral of the samples at every point is calculated by assuming a quadratic
+    relationship between each point and the two adjacent points.
+
+    Parameters
+    ----------
+    y : array_like
+        Values to integrate.
+    x : array_like, optional
+        The coordinate to integrate along. Must be monotonically increasing.
+        If None (default), use spacing `dx` between consecutive elements in `y`.
+    dx : scalar | array_like, optional
+        Spacing between elements of `y`. Only used if `x` is None. Can either be
+        a float, or an array with the same shape as `y`, but size 1 along `axis`.
+    axis : int, optional
+        Specifies the axis to integrate along. Default is -1 (last axis).
+    initial : scalar | array_like, optional
+        If given, insert this value at the beginning of the returned result,
+        and add it to the rest of the result. Default is None, which means no
+        value at ``x[0]`` is returned and `res` has one element less than `y`
+        along the axis of integration. Can either be a float, or an array with
+        the same shape as `y`, but size 1 along `axis`.
+
+    Returns
+    -------
+    res : ndarray
+        The result of cumulative integration of `y` along `axis`.
+        If `initial` is None, the shape is such that the axis of integration
+        has one less value than `y`. If `initial` is given, the shape is equal
+        to that of `y`.
+
+    See Also
+    --------
+    numpy.cumsum, numpy.cumprod
+    cumulative_trapezoid : cumulative integration using the composite trapezoidal rule
+    simpson : integrator for sampled data using the Composite Simpson's Rule
+    quad : adaptive quadrature using QUADPACK
+    romberg : adaptive Romberg quadrature
+    quadrature : adaptive Gaussian quadrature
+    fixed_quad : fixed-order Gaussian quadrature
+    dblquad : double integrals
+    tplquad : triple integrals
+    romb : integrators for sampled data
+    ode : ODE integrators
+    odeint : ODE integrators
+
+    Notes
+    -----
+
+    .. versionadded:: 1.11.0
+
+    For samples that are equally spaced the result is exact if the function
+    is a polynomial of order 3 or less. If the samples are not equally spaced,
+    then the result is exact only if the function is a polynomial of order 2
+    or less.
+
+    The composite Simpson's 1/3 method is outlined in [1]. A quadratic relationship
```

Comment:
We might want to add the mathematical formulation here.
```suggestion
    The composite Simpson's 1/3 method is outlined in [1]_. A quadratic relationship
```

Reply:
Added in [92fb1ef](https://github.com/scipy/scipy/pull/18151/commits/92fb1efb267ff8b402c9f693105051bdb9f8ff85)